test/test_testing.py

```diff
@@ -2706,3 +2717,465 @@ def test_stop_stopped_service(self):
         self.assertEqual('foo', foo_info.name)
         self.assertEqual(pebble.ServiceStartup.ENABLED, foo_info.startup)
         self.assertEqual(pebble.ServiceStatus.ACTIVE, foo_info.current)
+
+
+class _PebbleStorageAPIsTestMixin:
```

Comment:
I really like that this allows you to test against a real pebble backend and know that the testing harness acts like the actual runtime. I'm not sure how we want to integrate this into CI, etc, but I love the validation and ability to check that we aren't fooling people into thinking it acts in a certain way.

Reply:
I don't actually think it'd be hard to run this against real Pebble in CI. We'd just need to use the `setup-go` GitHub action, and then `go install github.com/canonical/pebble/cmd/pebble@latest` to build the Pebble binary.

If @Vultaire wants to take a stab at this, happy to review, otherwise I might do it on the next rainy day.

Reply:
We'll get a separate Github Issue raised for this and track progress there.